src/v2/Artsy/Relay/middleware/cache/cacheMiddleware.ts

```diff
@@ -0,0 +1,93 @@
+import { QueryResponseCache } from "relay-runtime"
+import { isFunction } from "lodash"
+import createLogger from "v2/Utils/logger"
+import { Cache, CacheConfig } from "./Cache"
+
+const logger = createLogger("v2/Artsy/middleware/cache/cacheMiddleware")
+
+interface CacheMiddlewareOpts extends CacheConfig {
+  onInit?: (cache: QueryResponseCache) => any
+  allowMutations?: boolean
+  allowFormData?: boolean
+  clearOnMutation?: boolean
+  cacheErrors?: boolean
+  disableServerSideCache?: boolean
+}
+
+/**
+ * Adapted from `react-relay-network-modern`.
+ * @see https://github.com/relay-tools/react-relay-network-modern/tree/master/src/middlewares
+ */
+export function cacheMiddleware(opts?: CacheMiddlewareOpts) {
+  const {
+    size,
+    ttl,
+    onInit,
+    allowMutations,
+    allowFormData,
+    clearOnMutation,
+    cacheErrors,
+    disableServerSideCache = true,
+  } = opts || {}
+
+  const cache = new Cache({
+    size: size || 100, // 100 requests
+    ttl: ttl || 15 * 60 * 1000, // 15 minutes
+    disableServerSideCache,
+  })
+
+  if (isFunction(onInit)) {
```

Comment:
This is a bit of a nitpick. If we rely on TypeScript here, there is no need to check if `onInit` is a function and this could be simplified to just:

```ts
if (onInit) {
  // ...
}
```

Reply:
Ah, this file was mostly copy paste from `react-relay-network-modern` with some light adaptations. Will update 👍